File: (7 kyu) Number of People in the Bus/(7 kyu) Number of People in the Bus.py
# #1
# from functools import reduce
# def number(stops):
#     nums = [a[0] - a[1] for a in stops]
#     return reduce(lambda a, b: a + b, nums)
#2
def number(bus_stops):
    return sum([a[0] - a[1] for a in bus_stops])

File: 7kyu/(7 kyu) Number of People in the Bus/(7 kyu) Number of People in the Bus.rs
// #1
// fn number(stops: &[(i32,i32)]) -> i32 {
//     stops.iter().map(|c| c.0 - c.1).sum()
// }
// #2
fn number(stops: &[(i32, i32)]) -> i32 {
    stops.iter().fold(0, |a, c| a + c.0 - c.1)
}
